### PR TITLE
add blacklist of properties that can force layout/paint

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

File: build-system/tasks/presubmit-checks.js [hunk 1]
```diff
@@ -35,9 +36,6 @@ var forbiddenTerms = {
   'XXX': '',
```
Comment (dvoytenko) [inline build-system/tasks/presubmit-checks.js]: No XXX

**Comment by erwinmombay** [comment]
@cramforce PTAL

File: build-system/tasks/presubmit-checks.js [hunk 2]
```diff
@@ -45,11 +43,65 @@ var forbiddenTerms = {
...
+  '\\.scrollWidth(?!_)': 'please use `getScrollWidth()` from viewport',
+  '\\.scrollHeight(?!_)': bannedTermsHelpString,
+  '\\.scrollLeft(?!_)': bannedTermsHelpString,
+  '\\.scrollTop(?!_)': bannedTermsHelpString,
+  '\\.focus(?!_)': bannedTermsHelpString,
+  '\\.computedRole(?!_)': bannedTermsHelpString,
+  '\\.computedName(?!_)': bannedTermsHelpString,
+  '\\.innerText(?!_)': bannedTermsHelpString,
+  '\\.getComputedStyle(?!_)': bannedTermsHelpString,
+  '\\.scrollX(?!_)': bannedTermsHelpString,
```
Comment (dvoytenko) [inline build-system/tasks/presubmit-checks.js]: Since we are here, also need pageYOffset and pageXOffset - those are just synonyms of scrollY/X

File: src/viewport.js [hunk 3]
```diff
@@ -179,7 +179,7 @@ export class Viewport {
   }
 
   /**
-   * Scrolls element into view much like Element.scrollIntoView does but
+   * Scrolls element into view much like Element.\/*OK*\/scrollIntoView does but
```
Comment (dvoytenko) [inline src/viewport.js]: check if this works and/or remove, preferably the later.

**Comment by dvoytenko** [comment]
LGTM

File: build-system/tasks/presubmit-checks.js (hunk 1 above)
Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: done.

File: build-system/tasks/presubmit-checks.js (hunk 2 above)
Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: done.

File: src/viewport.js (hunk 3 above)
Comment (erwinmombay) [inline src/viewport.js]: removed and just added a space.

File: build-system/tasks/presubmit-checks.js [hunk 4]
```diff
@@ -45,11 +42,67 @@ var forbiddenTerms = {
...
   'requestFileSystem': '',
-  '\\.scrollTop': '',
   'webkitRequestFileSystem': '',
   'debugger': '',
 };
 
+var bannedTermsHelpString = 'Please review viewport.js for a helper method or'
+  'mark with `/*OK*/` or `/*REVIEW*/` and consult the AMP team.';
+
+// Most of the forbidden property/method access banned on the
```
Comment (cramforce) [inline build-system/tasks/presubmit-checks.js]: How about just making this part of the help string?

**Comment by cramforce** [comment]
LGTM

:ship:

File: build-system/tasks/presubmit-checks.js (hunk 4 above)
Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: done.

thought it might be too wordy, but it wasn't too bad when i tested it out on the command line.

**Comment by erwinmombay** [comment]
:pray: to make the linter happy
